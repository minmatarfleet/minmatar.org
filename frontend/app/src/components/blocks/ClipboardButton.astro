---
import { getLangFromUrl, useTranslations } from '@i18n/utils';

const lang = getLangFromUrl(Astro.url);
const t = useTranslations(lang);

import Button from '@components/blocks/Button.astro';
import ClipboardIcon from '@components/icons/buttons/ClipboardIcon.astro';

interface Props {
    id: string;
    alert_prefix?: string;
    [propName: string]: any;
}

const {
    id,
    alert_prefix,
    ...attributes
} = Astro.props

const onclick_attr = `copyFitToClipboard(${ alert_prefix ? `'${id}', '${alert_prefix}'` : `'${id}'` })`
const Textarea = 'textarea';
---

<Button onclick={onclick_attr} size='sm' {...attributes}>
    <ClipboardIcon slot="icon" />
    {t('copy')}
</Button>

<Textarea class="[ hidden ]" id={id} readonly><slot /></Textarea>