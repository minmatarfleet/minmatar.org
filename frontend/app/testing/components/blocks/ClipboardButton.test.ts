import { experimental_AstroContainer as AstroContainer } from "astro/container";
import { expect, test, vi } from "vitest";
import ClipboardButton from "@components/blocks/ClipboardButton.astro";

test("ClipboardButton defaults", async () => {
  const container = await AstroContainer.create();
  const result = await container.renderToString(ClipboardButton, {
    props: {
      id: 56,
      alert_prefix: "what_the_cluck_",
    },
    slots: {
      default: "slotted chick",
    },
  });

  expect(result).toMatchSnapshot();
});
