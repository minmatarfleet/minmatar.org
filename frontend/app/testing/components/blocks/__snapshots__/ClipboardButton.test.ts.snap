// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ClipboardButton defaults 1`] = `
"<button x-on:click.stop.prevent="copyFitToClipboard('56', 'what_the_cluck_')" x-data="{
        button_text: 'Copy',
    }" x-on:click.throttle.750ms="button_text = 'Copied!'; setTimeout(() => button_text = 'Copy', 2000)" x-init="$el.removeAttribute(&#34;disabled&#34;)" disabled data-size="sm" style="--button_background_color: var(--fleet-red); --button_color: var(--fleet-yellow)" data-astro-cid-r5e2wo56="true" class="[ button ]"> <div class="flex-inline" data-astro-cid-r5e2wo56="true" data-astro-cid-ud5fqf5k style="--flex_inline_gap: var(--space-xs);--flex_inline_justification: center;">  <svg width="24" height="24" viewBox="0 0 16 20" fill="none" xmlns="http://www.w3.org/2000/svg"> <path d="M6 0H12V2H16V20H0V2H4V0H6ZM12 4V6H4V4H2V18H14V4H12ZM10 4V2H6V4H10Z" fill="currentColor"></path> </svg> <span x-text="button_text">Copy</span>   </div>   </button>  <textarea class="[ hidden ]" id="56" readonly>slotted chick</textarea>"
`;
